core/common/src/SkyBox.ts

```diff
@@ -91,6 +91,8 @@ export type SkyBoxImageProps = SkySphereImageProps | SkyCubeImageProps | { type?
  *
  * The color and exponent properties are unused if one or more texture images are supplied.
  *
+ * The color, exponent and texture properties are unused if an atmosphere is defined on the Environment.
```

Comment:
Isn't that *all* of the properties? Wouldn't it be clearer to say that we don't draw the skybox if atmosphere is enabled?

Reply:
Technically the SkyBox must be enabled currently for the atmosphere to draw, it just ignores all the other details other than "there is a skybox" when doing so

Reply:
Was that a deliberate decision? It seems odd to me that the atmosphere entirely replaces the skybox, but you must enable the skybox **and** the atmosphere to see the atmosphere.

Reply:
That's correct. The skyboxes would need to be specifically made to work with the atmosphere.
But unless we start doing business on another planet, there really would only need to be *one* atmosphere-compatible skybox since there's only one earth (or at least a choice of a few options). There wouldn't be much sense in everyone packing their own skybox to work with the atmosphere.

My main point was that it doesn't make sense to draw the atmospheric effect in the sky *not* in the skybox shader if we want to add that feature. Piggybacking off the existing skybox ViewportQuad is helpful.

I agree that its silly to not show the atmosphere in the sky if the skybox is disabled though.
To fix that, I can default to an empty skybox if the atmosphere is enabled and one isn't defined.

Reply:
> move atmospheric scattering in the sky to its own shader

Such an implementation decision shouldn't influence the public API we expose.

> Instead of tying the existing skybox texture to the atmosphere effect, though, we could eventually add another texture field to the atmosphere settings specifically for stars, etc.

This is where I am leaning.

Reply:
> Instead of tying the existing skybox texture to the atmosphere effect, though, we could eventually add another texture field to the atmosphere settings specifically for stars, etc.

I'll add these specific details to the enhancement work item so it gets remembered.

For the behavior of defining atmosphere with/without a skybox, are you leaning towards
- not drawing the atmospheric effect in the sky without a skybox
- forcing a (blank) skybox to be drawn when atmospheric effects are enabled
- or something different?

You left [a comment](https://github.com/iTwin/itwinjs-core/pull/3720#discussion_r992752405) a while back that implied the first option

Reply:
Please correct me if I misunderstand:
Whether or not the skybox is enabled in `Environment`, your implementation **needs** to draw a black-ish skybox when drawing the atmosphere for it to look right - right? It ignores whatever skybox settings might be defined on the `Environment`.
To me that's an implementation detail. From a user's perspective, if they turn on the skybox and the atmosphere, they only see a black sky with an atmosphere. So the simplest way for this to make sense to a user is: when you enable the atmosphere, the `Environment`'s skybox is ignored, whether or not it is also enabled.

Reply:
Your assessment is mostly correct; the only thing the atmospheric toggle doesn't ignore right now is if the environment's skybox is defined. It still needs to exist and be displayed to draw the sky atmosphere, but I agree that it's better to ignore that too.

I'll change the implementation to ignore skybox completely and adjust the documentation to match.